Formula/f/faircamp.rb

```diff
@@ -0,0 +1,30 @@
+class Faircamp < Formula
+  desc "Static site generator for audio producers"
+  homepage "https://codeberg.org/simonrepp/faircamp"
+  url "https://codeberg.org/simonrepp/faircamp/archive/0.9.2.tar.gz"
+  sha256 "15c826e22d8297223ae6c591d14ff79226682ef4db3911a1f349d639406295b7"
+  license "AGPL-3.0-or-later"
+
+  depends_on "ffmpeg" => :build
+  depends_on "opus" => :build
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on :macos # Linux builds aren't working at present
+  depends_on "vips"
+
+  def install
+    # libvips is a runtime dependency as it only ships a dynamic
+    # library with brew, its not discovered by default by cargo:
+    #   https://codeberg.org/simonrepp/faircamp/issues/45
+    ENV["RUSTFLAGS"] = `pkg-config --libs vips`.chomp
+    system "cargo", "install", *std_cargo_args, "--features", "libvips"
+  end
+
+  test do
+    # Faircamp doesn't presently ship with any cargo tests
+    # A minimal test that checks required dynamic libs are found
+    # and the binary notionally runs.
+    version_str = shell_output("#{bin}/faircamp --version").chomp
```

Comment:
we need some other tests than just checking version

Reply:
Can you elaborate on what tests would be required?

Reply:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks @SMillerDev - updated [here](https://github.com/Homebrew/homebrew-core/pull/153696/commits/05627035167884433b8ddeee95941c4d76ed73d2)